src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/reader/CatalogFileReader.java

```diff
@@ -0,0 +1,390 @@
+    }
+    
+    @Override
+    protected void beforeStep(StepExecution stepExecution) {
+        if(this.getXp() == null) {
+            setXp(this.xstream.getXStream());
+        }
+    }
+    
+    /**
+     * In strict mode the reader will throw an exception on {@link #open(org.springframework.batch.item.ExecutionContext)} if the input resource does
+     * not exist.
+     * 
+     * @param strict false by default
+     */
+    public void setStrict(boolean strict) {
+        this.strict = strict;
+    }
+
+    @Override
+    public void setResource(Resource resource) {
+        this.resource = resource;
+    }
+
+    /**
+     * @param fragmentRootElementName name of the root element of the fragment
+     */
+    public void setFragmentRootElementName(String fragmentRootElementName) {
+        setFragmentRootElementNames(new String[] { fragmentRootElementName });
+    }
+
+    /**
+     * @param fragmentRootElementNames list of the names of the root element of the fragment
+     */
+    public void setFragmentRootElementNames(String[] fragmentRootElementNames) {
+        this.fragmentRootElementNames = new ArrayList<QName>();
+        for (String fragmentRootElementName : fragmentRootElementNames) {
+            this.fragmentRootElementNames
+                    .add(parseFragmentRootElementName(fragmentRootElementName));
+        }
+    }
+
+    /**
+     * Ensure that all required dependencies for the ItemReader to run are provided after all properties have been set.
+     * 
+     * @see org.springframework.beans.factory.InitializingBean#afterPropertiesSet()
+     * @throws IllegalArgumentException if the Resource, FragmentDeserializer or FragmentRootElementName is null, or if the root element is empty.
+     * @throws IllegalStateException if the Resource does not exist.
+     */
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        Assert.notEmpty(fragmentRootElementNames, "The FragmentRootElementNames must not be empty");
+        for (QName fragmentRootElementName : fragmentRootElementNames) {
+            Assert.hasText(fragmentRootElementName.getLocalPart(),
+                    "The FragmentRootElementNames must not contain empty elements");
+        }
+    }
+    
+    /**
+     * Responsible for moving the cursor before the StartElement of the fragment root.
+     * 
+     * This implementation simply looks for the next corresponding element, it does not care about element nesting. You will need to override this
+     * method to correctly handle composite fragments.
+     * 
+     * @return <code>true</code> if next fragment was found, <code>false</code> otherwise.
+     * 
+     * @throws NonTransientResourceException if the cursor could not be moved. This will be treated as fatal and subsequent calls to read will return
+     *         null.
+     */
+    protected boolean moveCursorToNextFragment(XMLEventReader reader) {
+        try {
+            while (true) {
```

Comment:
Hum... so this code tries to read as if multiple xml fragments were in the file, but the backup configuration actually builds one file per layer?